core/vm/interpreter.go

```diff
@@ -112,15 +111,24 @@ func (in *Interpreter) Run(contract *Contract, input []byte) (ret []byte, err er
 	// as every returning call will return new data anyway.
 	in.returnData = nil
 
+	if in.intPool == nil {
+		in.intPool = poolOfIntPools.get()
+		defer func() {
+			poolOfIntPools.put(in.intPool)
+			in.intPool = nil
+		}()
+	}
+
 	// Don't bother with the execution if there's no code.
 	if len(contract.Code) == 0 {
 		return nil, nil
 	}
 
 	var (
-		op    OpCode        // current opcode
-		mem   = NewMemory() // bound memory
-		stack = newstack()  // local stack
+		op      OpCode         // current opcode
+		mem     = NewMemory()  // bound memory
+		stackIP = newIntPool() // intPool holding the stack
+		stack   = stackIP.pool // local stack
```

Comment:
I don't see the optimization here. A stack always starts out as empty, why would we want to use an intPool for it, the purpose of which is to not-be-empty.

Reply:
thed intpool is just a pointer to a stack; there's no optimization, just something that needs to be created anyway if the stack's int is pushed back onto the stack.

Reply:
please read my other comments. the stack should **not** be pushed into the pool

Reply:
ok, gotcha, in any case it should be `get()` and not `newIntPool`

Reply:
no, it shouldn't be either. it should be the original code. get will get you a non-empty stack, you want an empty one... as in the original code.

Reply:
yup, got it